### DISCUSSION TITLE
GH Code Scanning: 1 Analysis Not Found

### DISCUSSION THREAD

**Discussion by HACKERALERT** [community/community#24662] [Apps, API and Webhooks]
<p>I recently deleted my CodeQL analysis file (<code>/.github/workflows/codeql-analysis.yml</code>) but decided to add it back. After doing so, CodeQL scans my Go repository as expected for commits, but for pull requests, it shows:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/e/b/eb6d5197ff8a1f3493fafb932559bc66aabd4200.png" data-download-href="/uploads/short-url/xAGq1SMB3SKOwgx4mtfnRrBoyHu.png?dl=1" title="Screenshot (104)" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/48808396/181085920-b4d53e03-3490-419e-b555-4cb76b03dfb2.png" alt="Screenshot (104)" data-base62-sha1="xAGq1SMB3SKOwgx4mtfnRrBoyHu" width="690" height="391" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/e/b/eb6d5197ff8a1f3493fafb932559bc66aabd4200_2_690x391.png, https://aws1.discourse-cdn.com/github/optimized/3X/e/b/eb6d5197ff8a1f3493fafb932559bc66aabd4200_2_1035x586.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/e/b/eb6d5197ff8a1f3493fafb932559bc66aabd4200_2_1380x782.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/e/b/eb6d5197ff8a1f3493fafb932559bc66aabd4200_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">Screenshot (104)</span><span class="informations">1391×789 80.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>Clicking on the “Details”:<br>
<a href="https://imgur.com/a/CBVVEky" rel="noopener nofollow ugc">https://imgur.com/a/CBVVEky</a> (only one media is allowed for me)</p>
<p>My repository is a Go project and my <code>codeql-analysis.yml</code> currently is:</p>
<pre><code class="lang-auto">name: "CodeQL"
on:
  push:
    paths:
      - "src/XXXXXXXXX.go"
      - "src/go.mod"
      - "src/go.sum"
  pull_request:
    branches: [ main ]
jobs:
  analyze:
    name: Analyze
    runs-on: ubuntu-latest
    permissions:
      actions: read
      contents: read
      security-events: write
    strategy:
      fail-fast: false
      matrix:
        language: ['go']
    steps:
    - name: Checkout Repository
      uses: actions/checkout@v2
    - name: Initialize CodeQL
      uses: github/codeql-action/init@v1
      with:
        languages: ${{ matrix.language }}
    - name: Autobuild
      uses: github/codeql-action/autobuild@v1
    - name: Perform CodeQL Analysis
      uses: github/codeql-action/analyze@v1
</code></pre>
<p>Does anyone know how to get rid of the empty check? I have lots of info available and I can provide more info if it is needed. Thank you.</p>

**Comment by jsoref** [comment]
<p>I would cheat by creating a new empty repo, git pushing the same git repository contents into it, and then swapping the repo names (and eventually deleting the original repository).</p>
<p>This of course isn’t a great strategy if one has forks/stars, in which case I’d contact:</p>
<p><a href="https://support.github.com" class="onebox" target="_blank" rel="noopener nofollow ugc">https://support.github.com</a></p>

**Comment by HACKERALERT** [comment]
<p>Thanks for your reply! I’ve thought about that before, but I have around 300 stars and I don’t want to lose them, so this would probably not work for me. Thanks though.</p>

**Comment by adityasharad** [comment]
<p>Thanks for the detailed description of the problem you’re facing.<br>
Could you please try adjusting your workflow so that it runs on all pushes to your <code>main</code> branch, e.g.</p>
<pre><code class="lang-auto">on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
</code></pre>
<p>That should ensure that any PR against <code>main</code> will have a baseline of alerts to compare against, and resolve the error about the missing analysis. It’s possible that some existing commits on <code>main</code> don’t have corresponding analysis results (either because of the path filters or the period of time in which you turned off the analysis), so this suggestion should fix the problem for future commits and PRs. If the problem persists, please let us know here and we’ll be happy to investigate further.</p>

**Comment by HACKERALERT** [comment]
<p>Hmmm, I just tried that but it still shows a neutral check. For example, <a href="https://github.com/HACKERALERT/Picocrypt/pull/54/checks" class="inline-onebox" rel="noopener nofollow ugc">test by evanyiwensu · Pull Request #54 · HACKERALERT/Picocrypt · GitHub</a>. I’m not sure what the culprit is, but thanks for helping me make some progress :slight_smile:</p>

**Comment by adityasharad** [comment, marked as answer]
<p>If we look at <a href="https://github.com/HACKERALERT/Picocrypt/pull/54/checks" class="inline-onebox">test by evanyiwensu · Pull Request #54 · HACKERALERT/Picocrypt · GitHub</a>, we see the following:</p>
<pre><code class="lang-auto">| Category	| Base branch	| Pull request |
| .github/workflows/codeql-analysis.yml:analyze/language:python |	✔️  Found |	❓  Not found |
| .github/workflows/codeql-analysis.yml:analyze/language:go |	✔️  Found |	✔️  Found for 349a734 |
</code></pre>
<p>So it looks like things are working correctly for your Go analysis, and for Python there are existing analysis results on <code>main</code> (probably from earlier) but not on the PR (because the PR workflow doesn’t include Python anymore). This looks reasonable. You can leave things as they are without any significant impact to your checks/alerts. If you want to clean up the warning completely, use the <a href="https://docs.github.com/en/rest/reference/code-scanning#delete-a-code-scanning-analysis-from-a-repository">code scanning REST API</a> to delete the old analysis objects for Python, using the category described in that table.</p>

**Comment by HACKERALERT** [comment]
<p>Thank you for the clear and concise response! I will just leave it as is since it seems like there’s a lot to do to get rid of the notices. Thanks for your help.</p>

**Comment by LWVt** [comment]
<p>Hello<br>
Int (" hello-meni":);</p>
